### PR TITLE
ci: add check-docs workflow

### PR DESCRIPTION
**Pull request by yqlbu**
### Background

Apply markdown linting cuz most of the time, as human beings, we may make `typos` very easily.

Preview: https://github.com/yqlbu/dae/tree/ci-markdown-lint

### Checklist

- [x] The Pull Request has been fully tested

### Full changelog

- ci: add check-docs workflow
- ci: ONLY kick off build when targeting workflow file changes
- fix: apply syntax fix for all problematic `.md` files

### Issue reference

NA.

### CODE REVIEW

File: docs/commit-msg-guide.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Example `<type>` values:
 - `test`: (adding missing tests, refactoring tests; no production code change)
 - `chore`: (updating grunt tasks etc; no production code change, e.g. dependencies upgrade)
 - `perf`: (perfomance improvement change, e.g. better concurrency performance)
-- `ci`: (updating CI configuration files and scripts e.g. .github/workflows/*.yml )
+- `ci`: (updating CI configuration files and scripts e.g. .GitHub/workflows/\*.yml )
```
Comment (mzz2017) [inline docs/commit-msg-guide.md:39]:
```suggestion
- `ci`: (updating CI configuration files and scripts e.g. .gitHub/workflows/*.yml )
```

File: docs/getting-started/kernel-upgrade.md [hunk 2]
```diff
@@ -46,12 +46,12 @@ sudo reboot
 uname -r
 ```
 
-(DEBIAN ONLY): If you would like to upgrade to the latest Kernel (AGGRESSIVE UPGRADE), follow the commands below:
+(Debian ONLY): If you would like to upgrade to the latest Kernel (AGGRESSIVE UPGRADE), follow the commands below:
 
 > **Warning**
-> The latest Kernel that Debian officially supports is available in the `unstable release`. Debian Unstable (also known by its codename "Sid") is not strictly a release, but rather a rolling development version of the Debian distribution containing the latest packages that have been introduced into Debian. Upgrading to the latest Kernel might potentially introduce breaking changes to your system, so please do at your own risk.
+> The latest Kernel that Debian officially supports is available in the `unstable release`. Debian Unstable (also known by its codename "SID") is not strictly a release, but rather a rolling development version of the Debian distribution containing the latest packages that have been introduced into Debian. Upgrading to the latest Kernel might potentially introduce breaking changes to your system, so please do at your own risk.
```
Comment (mzz2017) [inline docs/getting-started/kernel-upgrade.md:52]: Why not 
```
**Warning**: xxxxx
```

Comment (yqlbu) [reply docs/getting-started/kernel-upgrade.md:52]: This is a good catch, I will fix it in a sec.

Comment (yqlbu) [reply docs/getting-started/kernel-upgrade.md]: Working like a charm - https://github.com/yqlbu/dae/blob/ci-markdown-lint/docs/getting-started/kernel-upgrade.md

**Review by mzz2017 — APPROVED** [review]

**Comment by mzz2017** [comment]
Good job!

**Pull request merged**